packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdkBridgeExt.kt

```diff
@@ -47,3 +57,72 @@ internal fun Map<*, *>.toWritableMap(): WritableNativeMap {
     }
     return map
 }
+
+/**
+ * Casts the ReadableArray to a List that strictly contains non-null objects of type HashMap.
+ * @throws TypeCastException if at least one of the list items is not of type HashMap.
+ */
+internal fun ReadableArray.toHashMapArrayList(): List<HashMap<*, *>> {
+    return this.toKotlinArrayList().toMutableList().mapNotNull {
+        if (it is HashMap<*, *> || it == null) {
+            it as HashMap<*, *>?
+        } else {
+            throw TypeCastException(
+                "Cannot convert ReadableArray to ArrayList of HashMap(s): " +
+                    "$it is not of type HashMap<*,*>",
+            )
+        }
+    }
+}
+
+/**
+ * Recursively converts the ReadableMap to a HashMap which only contains Kotlin stdlib objects,
+ * such as ArrayList, HashMap and the raw types.
+ * @throws IllegalArgumentException if the readable map is malformed (eg. it contains a HashMap
+ * or ArrayList, instead of ReadableMap and ReadableArray respectively).
+ */
+internal fun ReadableMap.toKotlinHashMap(): HashMap<String, Any> {
+    val hashMap: HashMap<String, Any> = this.toHashMap()
```

Comment:
type is already clear from the call name

```suggestion
    val map = this.toHashMap()
```